msgraph/directory_objects.go

```diff
@@ -0,0 +1,191 @@
+package msgraph
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/manicminer/hamilton/internal/utils"
+	"github.com/manicminer/hamilton/odata"
+)
+
+// DirectoryObjectsClient performs operations on Directory Objects (the base type for other objects such as users and groups)
+type DirectoryObjectsClient struct {
+	BaseClient Client
+}
+
+// NewDirectoryObjectsClient returns a new DirectoryObjectsClient.
+func NewDirectoryObjectsClient(tenantId string) *DirectoryObjectsClient {
+	return &DirectoryObjectsClient{
+		BaseClient: NewClient(Version10, tenantId),
+	}
+}
+
+// Get retrieves a DirectoryObject.
+func (c *DirectoryObjectsClient) Get(ctx context.Context, id string, query odata.Query) (*DirectoryObject, int, error) {
+	resp, status, _, err := c.BaseClient.Get(ctx, GetHttpRequestInput{
+		ConsistencyFailureFunc: RetryOn404ConsistencyFailureFunc,
+		ValidStatusCodes:       []int{http.StatusOK},
+		Uri: Uri{
+			Entity:      fmt.Sprintf("/directoryObjects/%s", id),
+			Params:      query.Values(),
+			HasTenantId: true,
+		},
+	})
+	if err != nil {
+		return nil, status, fmt.Errorf("DirectoryObjects.BaseClient.Get(): %v", err)
+	}
+	defer resp.Body.Close()
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, status, fmt.Errorf("io.ReadAll(): %v", err)
+	}
+	var directoryObject DirectoryObject
+	if err := json.Unmarshal(respBody, &directoryObject); err != nil {
+		return nil, status, fmt.Errorf("json.Unmarshal(): %v", err)
+	}
+	return &directoryObject, status, nil
```

Comment:
i'd personlly like to see a little more whitespace here
```suggestion
	})
	if err != nil {
		return nil, status, fmt.Errorf("DirectoryObjects.BaseClient.Get(): %v", err)
	}
	defer resp.Body.Close()
	
	respBody, err := io.ReadAll(resp.Body)
	if err != nil {
		return nil, status, fmt.Errorf("io.ReadAll(): %v", err)
	}
	
	var directoryObject DirectoryObject
	if err := json.Unmarshal(respBody, &directoryObject); err != nil {
		return nil, status, fmt.Errorf("json.Unmarshal(): %v", err)
	}
	
	return &directoryObject, status, nil
```

Reply:
Good call, these grew organically and never got truly tidied up.